SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/EventTracingTests/EventTracingTest.cpp

```diff
@@ -190,20 +190,17 @@ void AEventTracingTest::GatherDataFromFile(const FString& FilePath)
 	TUniquePtr<Io_Stream, StreamDeleter> Stream;
 	Stream.Reset(Io_CreateFileStream(TCHAR_TO_ANSI(*FilePath), Io_OpenMode::IO_OPEN_MODE_READ));
 
-	uint32_t BytesToRead = 1;
-	int8_t ReturnCode = 1;
-	while (BytesToRead != 0 && ReturnCode == 1)
+	int64_t BytesRead = 1;
+	while (BytesRead > 0)
 	{
-		BytesToRead = Trace_GetNextSerializedItemSize(Stream.Get());
-		if (BytesToRead == 0)
-		{
-			break;
-		}
-
 		Trace_Item* Item = Trace_Item_GetThreadLocal();
-		if (BytesToRead != 0)
+		BytesRead = Trace_DeserializeItemFromStream(Stream.Get(), Item);
```

Comment:
I guess we expect this to return 0 when it has finished reading?

Reply:
Yup - the new docstring for Trace_DeserializeItemFromStream: 
```
 * Returns a positive value (the number of bytes read from the stream) if the next serialized item
 * was successfully deserialized into the provided item.
 * Returns 0 if the stream contains no more data or only a partial (serialized) trace item.
 * In this case, you can write further data to the stream and try to call
 * Trace_DeserializeItemFromStream again.
 * Returns -1 if there was an error during deserialization. Call Trace_GetLastError to get the
 * associated error message.
 ```

Reply:
Nice!